adapters/adyoulike/adyoulike.go

```diff
@@ -0,0 +1,159 @@
+package adyoulike
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+
+	"github.com/buger/jsonparser"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+)
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	return &AdYouLikeAdapter{
+		endpoint: config.Endpoint,
+	}, nil
+}
+
+type AdYouLikeAdapter struct {
+	endpoint string
+}
+
+func (adapter *AdYouLikeAdapter) MakeRequests(
+	openRTBRequest *openrtb.BidRequest,
+	reqInfo *adapters.ExtraRequestInfo,
+) (
+	requestsToBidder []*adapters.RequestData,
+	errs []error,
+) {
+	var err error
+	var tagID string
+
+	if len(openRTBRequest.Imp) > 0 {
+		var imp = &openRTBRequest.Imp[0]
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, newBadInputError(err.Error()))
+		}
+
+		var ttxExt openrtb_ext.ExtImpAdyoulike
+		if err := json.Unmarshal(bidderExt.Bidder, &ttxExt); err != nil {
+			errs = append(errs, newBadInputError(err.Error()))
+		}
+	}
+
+	if len(openRTBRequest.Imp) == 0 {
+		errs = append(errs, newBadInputError("No impression in the bid request"))
+	}
+
+	reqCopy := *openRTBRequest
+	reqCopy.Imp = []openrtb.Imp{}
+	for ind, imp := range openRTBRequest.Imp {
```

Comment:
The `range` keyword creates a shallow copy of the elements of the array that will be limited to the scope of the `for` loop. Given that `TagID` is not a pointer, maybe we could modify every `imp`, then modify, then append:
```
55 -   for ind, imp := range openRTBRequest.Imp {
55 +   for _, imp := range openRTBRequest.Imp {
56 -       reqCopy.Imp = append(reqCopy.Imp, imp)
57
58 -       tagID, err = jsonparser.GetString(reqCopy.Imp[ind].Ext, "bidder", "placement")
   +       tagID, err = jsonparser.GetString(imp.Ext, "bidder", "placement")
59         if err != nil {
60             errs = append(errs, err)
61             continue
62         }
63
64 -       reqCopy.Imp[ind].TagID = tagID
64 +       imp.TagID = tagID
   +
   +       reqCopy.Imp = append(reqCopy.Imp, imp)
65     }
adapters/adyoulike/adyoulike.go
```